usr/lib/pkcs11/opencryptoki_tok.map

```diff
@@ -0,0 +1,69 @@
+OPENCRYPTOKI_TOK_3.10 {
+    global:
+        SC_CancelFunction;
+        SC_CloseAllSessions;
+        SC_CloseSession;
+        SC_CopyObject;
+        SC_CreateObject;
+        SC_Decrypt;
+        SC_DecryptDigestUpdate;
+        SC_DecryptFinal;
+        SC_DecryptInit;
+        SC_DecryptUpdate;
+        SC_DecryptVerifyUpdate;
+        SC_DeriveKey;
+        SC_DestroyObject;
+        SC_Digest;
+        SC_DigestEncryptUpdate;
+        SC_DigestFinal;
+        SC_DigestInit;
+        SC_DigestKey;
+        SC_DigestUpdate;
+        SC_Encrypt;
+        SC_EncryptFinal;
+        SC_EncryptInit;
+        SC_EncryptUpdate;
+        SC_Finalize;
+        SC_FindObjects;
+        SC_FindObjectsFinal;
+        SC_FindObjectsInit;
+        SC_GenerateKey;
+        SC_GenerateKeyPair;
+        SC_GenerateRandom;
+        SC_GetAttributeValue;
+        SC_GetFunctionStatus;
+        SC_GetMechanismInfo;
+        SC_GetMechanismList;
+        SC_GetObjectSize;
+        SC_GetOperationState;
+        SC_GetSessionInfo;
+        SC_GetTokenInfo;
+        SC_InitPIN;
+        SC_InitToken;
+        SC_Login;
+        SC_Logout;
+        SC_OpenSession;
+        SC_SeedRandom;
+        SC_SetAttributeValue;
+        SC_SetFunctionList;
+        SC_SetOperationState;
+        SC_SetPIN;
+        SC_Sign;
+        SC_SignEncryptUpdate;
+        SC_SignFinal;
+        SC_SignInit;
+        SC_SignRecover;
+        SC_SignRecoverInit;
+        SC_SignUpdate;
+        SC_UnwrapKey;
+        SC_Verify;
+        SC_VerifyFinal;
+        SC_VerifyInit;
+        SC_VerifyRecover;
+        SC_VerifyRecoverInit;
+        SC_VerifyUpdate;
+        SC_WaitForSlotEvent;
+        SC_WrapKey;
+        ST_Initialize;
+    local: *;
+};
```

Comment:
Patrick, do I understand this right ? This is the interface between ock base and the tokens ?
So every token builds some glue code around the token struct which on the upper side has all these functions ?

Reply:
yes